### PR TITLE
Add arrow markers to visualize goal poses of entities

### PR DESCRIPTION
**Pull request by Utaro-M**
## Types of PR

- [x] New Features
- [ ] Upgrade of existing features
- [ ] Bugfix

## Description
Add arrow markers to visualize goal poses of entities

### CODE REVIEW

File: openscenario/openscenario_msgs/msg/EntityStatusWithTrajectory.msg [hunk 1]
```diff
@@ -2,6 +2,6 @@ float64 time 0
 string name
 openscenario_msgs/EntityStatus status
 openscenario_msgs/WaypointsArray waypoint
-
+geometry_msgs/Pose[] goalposes
```
Comment (hakuturu583) [inline openscenario/openscenario_msgs/msg/EntityStatusWithTrajectory.msg:5]: can you rename to goal_poses ?
goalposes was detected as typo.

File: openscenario/openscenario_visualization/include/openscenario_visualization/openscenario_visualization_component.hpp [hunk 2]
```diff
@@ -108,8 +108,10 @@ class OpenscenarioVisualizationComponent : public rclcpp::Node
    * @param obstacle obstacles in waypoint
    * @return const visualization_msgs::msg::MarkerArray markers which describes entity bounding box and it's status.
    */
+  int goalposes_max_size = 0;
   const visualization_msgs::msg::MarkerArray generateMarker(
     const openscenario_msgs::msg::EntityStatus & status,
+    const std::vector<geometry_msgs::msg::Pose> & goalposes,
```
Comment (hakuturu583) [inline openscenario/openscenario_visualization/include/openscenario_visualization/openscenario_visualization_component.hpp:114]: can you rename to goal_poses ?
goalposes was detected as typo in other line.

File: openscenario/openscenario_visualization/src/openscenario_visualization_component.cpp [hunk 3]
```diff
@@ -90,7 +90,7 @@ void OpenscenarioVisualizationComponent::entityStatusCallback(
   }
   for (const auto & data : msg->data) {
     auto marker_array =
-      generateMarker(data.status, data.waypoint, data.obstacle, data.obstacle_find);
+      generateMarker(data.status, data.goalposes, data.waypoint, data.obstacle, data.obstacle_find);
```
Comment (hakuturu583) [inline openscenario/openscenario_visualization/src/openscenario_visualization_component.cpp:93]: can you rename to goal_poses ?
goalposes was detected as typo in other line.

File: openscenario/openscenario_visualization/src/openscenario_visualization_component.cpp [hunk 4]
```diff
@@ -118,6 +118,7 @@ const visualization_msgs::msg::MarkerArray OpenscenarioVisualizationComponent::g
 
 const visualization_msgs::msg::MarkerArray OpenscenarioVisualizationComponent::generateMarker(
   const openscenario_msgs::msg::EntityStatus & status,
+  const std::vector<geometry_msgs::msg::Pose> & goalposes,
```
Comment (hakuturu583) [inline openscenario/openscenario_visualization/src/openscenario_visualization_component.cpp:121]: can you rename to goal_poses ?
goalposes was detected as typo in other line.

File: simulation/traffic_simulator/include/traffic_simulator/behavior/route_planner.hpp [hunk 5]
```diff
@@ -35,6 +35,7 @@ class RoutePlanner
   std::vector<std::int64_t> getRouteLanelets(
     openscenario_msgs::msg::LaneletPose entity_lanelet_pose, double horizon = 100);
   void cancelGoal();
+  std::vector<openscenario_msgs::msg::LaneletPose> getGoalposes();
```
Comment (hakuturu583) [inline simulation/traffic_simulator/include/traffic_simulator/behavior/route_planner.hpp:38]: Can you change to getGoalPoses ?

File: simulation/traffic_simulator/include/traffic_simulator/entity/entity_base.hpp [hunk 6]
```diff
@@ -93,6 +93,8 @@ class EntityBase
 
   virtual auto getWaypoints() -> const openscenario_msgs::msg::WaypointsArray = 0;
 
+  virtual auto getGoalposes() ->  std::vector<openscenario_msgs::msg::LaneletPose> =0;
```
Comment (hakuturu583) [inline simulation/traffic_simulator/include/traffic_simulator/entity/entity_base.hpp:96]: Can you change to getGoalPoses ?

File: simulation/traffic_simulator/include/traffic_simulator/entity/entity_manager.hpp [hunk 7]
```diff
@@ -275,6 +275,10 @@ class EntityManager
 
   auto getWaypoints(const std::string & name) -> openscenario_msgs::msg::WaypointsArray;
 
+  void getGoalposes(const std::string & name, std::vector<openscenario_msgs::msg::LaneletPose> & goals);
```
Comment (hakuturu583) [inline simulation/traffic_simulator/include/traffic_simulator/entity/entity_manager.hpp:278]: Can you change to getGoalPoses ?

File: simulation/traffic_simulator/include/traffic_simulator/entity/entity_manager.hpp [hunk 8]
```diff
@@ -275,6 +275,10 @@ class EntityManager
...
+
+  void getGoalposes(const std::string & name, std::vector<geometry_msgs::msg::Pose> & goals);
```
Comment (hakuturu583) [inline simulation/traffic_simulator/include/traffic_simulator/entity/entity_manager.hpp:280]: Can you change to getGoalPoses ?

File: simulation/traffic_simulator/include/traffic_simulator/entity/misc_object_entity.hpp [hunk 9]
```diff
@@ -39,6 +39,8 @@ class MiscObjectEntity : public EntityBase
     return result;
   }
 
+  std::vector<openscenario_msgs::msg::LaneletPose> getGoalposes() override { return {}; } //return {}?
```
Comment (hakuturu583) [inline simulation/traffic_simulator/include/traffic_simulator/entity/misc_object_entity.hpp:42]: Can you change to getGoalPoses ?

File: simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp [hunk 10]
```diff
@@ -94,6 +94,8 @@ class PedestrianEntity : public EntityBase
 
   boost::optional<openscenario_msgs::msg::Obstacle> getObstacle() override { return boost::none; }
 
+  std::vector<openscenario_msgs::msg::LaneletPose> getGoalposes() override { return route_planner_ptr_->getGoalposes(); }
```
Comment (hakuturu583) [inline simulation/traffic_simulator/include/traffic_simulator/entity/pedestrian_entity.hpp:97]: Can you change to getGoalPoses ?

File: simulation/traffic_simulator/include/traffic_simulator/entity/vehicle_entity.hpp [hunk 11]
```diff
@@ -118,6 +118,8 @@ class VehicleEntity : public EntityBase
     }
   }
 
+  std::vector<openscenario_msgs::msg::LaneletPose> getGoalposes() override { return route_planner_ptr_->getGoalposes(); }
```
Comment (hakuturu583) [inline simulation/traffic_simulator/include/traffic_simulator/entity/vehicle_entity.hpp:121]: Can you change to getGoalPoses ?

File: simulation/traffic_simulator/src/entity/entity_manager.cpp [hunk 12]
```diff
@@ -363,6 +363,26 @@ auto EntityManager::getWaypoints(const std::string & name) -> openscenario_msgs:
   return entities_.at(name)->getWaypoints();
 }
 
+void EntityManager::getGoalposes(const std::string & name, std::vector<openscenario_msgs::msg::LaneletPose> & goals)
```
Comment (hakuturu583) [inline simulation/traffic_simulator/src/entity/entity_manager.cpp:366]: Can you change to getGoalPoses ?

File: simulation/traffic_simulator/src/entity/entity_manager.cpp [hunk 13]
```diff
@@ -363,6 +363,26 @@ auto EntityManager::getWaypoints(const std::string & name) -> openscenario_msgs:
...
+{
+  if (current_time_ < 0) {
+    goals = std::vector<openscenario_msgs::msg::LaneletPose>();
+  }
+  goals = entities_.at(name)->getGoalposes();
+}
+
+void EntityManager::getGoalposes(const std::string & name, std::vector<geometry_msgs::msg::Pose> & goals)
```
Comment (hakuturu583) [inline simulation/traffic_simulator/src/entity/entity_manager.cpp:374]: Can you change to getGoalPoses ?

**Review by hakuturu583 — CHANGES_REQUESTED** [review]
Typo was detected, so please fix goal poses variable name.

**Review by Utaro-M — COMMENTED** [review]
修正してpushしました。
よろしくお願いします。

File: simulation/traffic_simulator/src/behavior/route_planner.cpp [hunk 14]
```diff
@@ -95,6 +95,18 @@ void RoutePlanner::cancelGoal(const openscenario_msgs::msg::LaneletPose & entity
   }
 }
 
+std::vector<openscenario_msgs::msg::LaneletPose> RoutePlanner::getGoalPoses()
+{
+  std::vector<openscenario_msgs::msg::LaneletPose> goalposes;
```
Comment (hakuturu583) [inline simulation/traffic_simulator/src/behavior/route_planner.cpp:100]: goalposes was detected as typo in spellcheck.
Please modify to goal_poses

**Review by hakuturu583 — CHANGES_REQUESTED** [review]
Some word was detected as typo in spell checker, so please change wordings.

File: docs/ReleaseNotes.md [hunk 15]
```diff
@@ -1,6 +1,7 @@
 # Release Notes
 
 ## Difference between the latest release and master
+- Add arrow markers to visualize goal poses of entities.
```
Comment (hakuturu583) [inline docs/ReleaseNotes.md:4]: please adding your github account like

> Add a new metrics module which detects vel, acc and jerk out of range. (Contribution by kyabe2718).

**Review by hakuturu583 — CHANGES_REQUESTED** [review]

**Comment by hakuturu583** [comment]
> 修正してpushしました。
> よろしくお願いします。

please use english in Pull request.

File: openscenario/openscenario_visualization/src/openscenario_visualization_component.cpp [hunk 16]
```diff
@@ -135,6 +136,76 @@ const visualization_msgs::msg::MarkerArray OpenscenarioVisualizationComponent::g
       color = color_utils::makeColorMsg("lightskyblue", 0.99);
       break;
   }
+
+  if (goal_poses.size() != 0) {
+    goal_poses_max_size = std::max(goal_poses_max_size, int(goal_poses.size()));
+    for (std::vector<geometry_msgs::msg::Pose>::size_type i = 0; i < unsigned(goal_poses_max_size);
+         i++) {
+      if (i < goal_poses.size()) {
+        visualization_msgs::msg::Marker goalpose_marker;
```
[the remark this answers is not in the document]
Comment (hakuturu583) [reply openscenario/openscenario_visualization/src/openscenario_visualization_component.cpp:145]: It seems, this miss-spell was not fixed yet.

File: openscenario/openscenario_visualization/src/openscenario_visualization_component.cpp [hunk 17]
```diff
@@ -135,6 +136,76 @@ const visualization_msgs::msg::MarkerArray OpenscenarioVisualizationComponent::g
...
+        goalpose_marker.header.frame_id = "map";
+        goalpose_marker.header.stamp = stamp;
+        goalpose_marker.ns = status.name;
```
[the remark this answers is not in the document]
Comment (hakuturu583) [reply openscenario/openscenario_visualization/src/openscenario_visualization_component.cpp:148]: It seems, this miss-spell was not fixed yet.

File: openscenario/openscenario_visualization/src/openscenario_visualization_component.cpp [hunk 18]
```diff
@@ -135,6 +136,76 @@ const visualization_msgs::msg::MarkerArray OpenscenarioVisualizationComponent::g
...
       break;
   }
+
+  if (goal_poses.size() != 0) {
+    goal_poses_max_size = std::max(goal_poses_max_size, int(goal_poses.size()));
+    for (std::vector<geometry_msgs::msg::Pose>::size_type i = 0; i < unsigned(goal_poses_max_size);
+         i++) {
+      if (i < goal_poses.size()) {
+        visualization_msgs::msg::Marker goalpose_marker;
+        goalpose_marker.header.frame_id = "map";
```
[the remark this answers is not in the document]
Comment (hakuturu583) [reply openscenario/openscenario_visualization/src/openscenario_visualization_component.cpp:146]: It seems, this miss-spell was not fixed yet.

File: openscenario/openscenario_visualization/src/openscenario_visualization_component.cpp [hunk 19]
```diff
@@ -135,6 +136,76 @@ const visualization_msgs::msg::MarkerArray OpenscenarioVisualizationComponent::g
...
+        goalpose_marker.header.stamp = stamp;
```
[the remark this answers is not in the document]
Comment (hakuturu583) [reply openscenario/openscenario_visualization/src/openscenario_visualization_component.cpp:147]: It seems, this miss-spell was not fixed yet.

File: simulation/traffic_simulator/src/behavior/route_planner.cpp (hunk 14 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/src/behavior/route_planner.cpp:100]: It seems, this miss-spell was not fixed yet.

File: simulation/traffic_simulator/src/behavior/route_planner.cpp [hunk 20]
```diff
@@ -95,6 +95,18 @@ void RoutePlanner::cancelGoal(const openscenario_msgs::msg::LaneletPose & entity
...
+  goalposes = {};
```
Comment (hakuturu583) [inline simulation/traffic_simulator/src/behavior/route_planner.cpp:101]: It seems, this miss-spell was not fixed yet.

File: simulation/traffic_simulator/src/behavior/route_planner.cpp [hunk 21]
```diff
@@ -95,6 +95,18 @@ void RoutePlanner::cancelGoal(const openscenario_msgs::msg::LaneletPose & entity
...
+  auto waypoint_queue_tmp_ = waypoint_queue_;
+  for (int i = 0; !waypoint_queue_tmp_.empty(); i++) {
+    goalposes.push_back(waypoint_queue_tmp_.front());
```
Comment (hakuturu583) [inline simulation/traffic_simulator/src/behavior/route_planner.cpp:104]: It seems, this miss-spell was not fixed yet.

File: simulation/traffic_simulator/src/behavior/route_planner.cpp [hunk 22]
```diff
@@ -95,6 +95,18 @@ void RoutePlanner::cancelGoal(const openscenario_msgs::msg::LaneletPose & entity
...
+    waypoint_queue_tmp_.pop();
+  }
+  return goalposes;
```
Comment (hakuturu583) [inline simulation/traffic_simulator/src/behavior/route_planner.cpp:107]: It seems, this miss-spell was not fixed yet.

**Review by hakuturu583 — CHANGES_REQUESTED** [review]
Miss-spelled worlds are still exists.
Please fix typos.

**Review by hakuturu583 — APPROVED** [review]
spell check will be cleaned later, so I merge this PR first.

**Pull request merged**